zerocopy-derive/src/lib.rs

```diff
@@ -55,6 +55,45 @@ macro_rules! try_or_print {
     };
 }
 
+/// Inline size assertion annotations
+/// Any non-generic type can be check if the size is as expected or not in compile time
+/// # Implementation
+/// ```
+/// # use zerocopy_derive::inline_assert_size_eq;
+/// # use static_assertions;
+/// #[inline_assert_size_eq(4)]
+/// struct MyStruct {
+///     val: i32
+/// }
+/// ```
+/// # Example that will fail
+/// ```compile_fail
+/// # use zero_derive::inline_assert_size_eq;
+/// # use static_assertions;
+/// #[inline_assert_size_eq(1)]
+/// struct MyStruct {
+///     val: i32
+/// }
+/// ```
+#[proc_macro_attribute]
+pub fn inline_assert_size_eq(
```

Comment:
I would prefer to call this `assert_size_eq`, and drop the `inline_` prefix.

Reply:
There is `assert_size_eq` in [`macro_util.rs`](https://github.com/google/zerocopy/blob/main/src/macro_util.rs) which check if the size are equal between two types. I think we need another name. How about `assert_size_eq_val`?

Reply:
The items in `macro_util.rs` are internal implementation details of public macros. We can rename these internal implementation details.

Reply:
Done